specification/ai/Azure.AI.Projects/agents/tools/tool_resources.tsp

```diff
@@ -92,13 +92,37 @@ model IndexResource {
    * An index connection id in an IndexResource attached to this agent.
    */
   @encodedName("application/json", "index_connection_id")
-  indexConnectionId: string;
+  indexConnectionId?: string;
 
   /**
    * The name of an index in an IndexResource attached to this agent.
    */
   @encodedName("application/json", "index_name")
-  indexName: string;
+  indexName?: string;
+
+  /**
+   * Specifies what kind of search to do
+   */
+  @encodedName("application/json", "query_type")
+  queryType?: QueryTypes;
+
+  /**
+   * Number of documents to retrieve from search and present to the model
+   */
+  @encodedName("application/json", "top_k")
+  topK?: int64;
+
+  /**
+   * OData filter string configured by the client
+   */
+  @encodedName("application/json", "filter")
```

Comment:
encoded name is not needed here either, same comment applies everywhere the model property is the same as the encoded name. Otherwise the compiler will complain